### ISSUE TITLE
Questions about manually passing gradient

### ISSUE THREAD

**Issue by SkidanovAlex** [dmlc/mxnet#2455]
Hi,
I am building multiple models that logically are connected to each other, but in reality I want to train them separately and manually pass the gradient back. In other words, I define two models A and B, while logically I have a model

A -> B

Where one of the inputs to B is logically connected to an output of A. For simplicity let's assume that both A and B are simple logistic regression modules, and I want to connect them into a two layer neural network.

I first train B, and now I want to pass the gradient back to A. I know how to get gradient of the loss function with respect to the weights, my question is: is there any way to get the gradient with respect to the inputs, that would have been used for back propagation if it was not the input layer but rather an intermediate layer? In my example, I would want to know the error of the inputs to B to use it to compute gradients of weights in A.

One solution I found is to create a new identity layer, that just passes the error back and remembers it. Is there a more straightforward way to do it?

A good example of what I want is here in Torch:

https://github.com/karpathy/char-rnn/blob/master/train.lua

Look at lines 284-292, where he runs LSTM modules one by one and passes gradient back manually.

**Comment by tqchen** [comment]
This seems can be better achieved with two graphs, see the example usage of Modules in the DCGAN example

https://github.com/dmlc/mxnet/blob/master/example/gan/dcgan.py

**Comment by SkidanovAlex** [comment]
Thanks, it has the answer:

`modD.get_input_grads()`
